datafusion/physical-expr-functions-aggregate/src/aggregate.rs

```diff
@@ -15,34 +15,25 @@
 // specific language governing permissions and limitations
 // under the License.
 
-use std::fmt::Debug;
-use std::{any::Any, sync::Arc};
-
 use arrow::datatypes::{DataType, Field, Schema, SchemaRef};
-
-use datafusion_common::exec_err;
 use datafusion_common::{internal_err, not_impl_err, DFSchema, Result};
 use datafusion_expr::expr::create_function_physical_name;
-use datafusion_expr::function::StateFieldsArgs;
-use datafusion_expr::type_coercion::aggregates::check_arg_count;
-use datafusion_expr::utils::AggregateOrderSensitivity;
+use datafusion_expr::AggregateUDF;
```

Comment:
The blocker for us to move this crate into `common-level` crate `functions-aggregate-common` is `AggregateUDF`. It has `fn call() -> Expr` that tightly coupled with `Expr`.

Reply:
Maybe we should remove `call()`, but create another function like
`aggregate_expr(sum_udaf(), vec![args]) -> Expr` 
And, `aggregate_physical_expr(sum_udaf(), vec![physical_args]) -> Arc<dyn AggregateExpr>>` similarly. 🤔